atomicapp/providers/kubernetes.py

```diff
@@ -38,227 +44,234 @@ class KubernetesProvider(Provider):
 
-    def _find_kubectl(self, prefix=""):
-        """Determine the path to the kubectl program on the host.
-        1) Check the config for a provider_cli in the general section
-           remember to add /host prefix
-        2) Search /usr/bin:/usr/local/bin
+            Params:
+                If any provider specific parameters have been provided,
+                load the configuration through the answers.conf file
 
-        Use the first valid value found
+            Default configuration:
+                If no paramaters have been provider, as well as --provider-config,
+                use the default .kube/config location and load the associated configuration.
+            '''
+            if self.config_file:
+                self.api = Client(KubeConfig.from_file(self.config_file), "kubernetes")
+            elif self._check_required_params():
+                self.api = Client(self._from_required_params(), "kubernetes")
+            else:
+                config_location = os.path.join(Utils.getRoot(), Utils.getUserHome().strip('/'), '.kube/config')
+                self.api = Client(KubeConfig.from_file(config_location), "kubernetes")
+
+            # Check if the namespace that the app is being deployed to is available
+            self._check_namespaces()
+
+    def _check_required_params(self):
+        '''
+        This checks to see if any parameters associated to the Kubernetes provider are passed
+        '''
+
+        # Initialize the values
+        result = {PROVIDER_API_KEY: self.provider_api,
+                  PROVIDER_AUTH_KEY: self.provider_auth,
+                  PROVIDER_TLS_VERIFY_KEY: self.provider_tls_verify,
+                  PROVIDER_CA_KEY: self.provider_ca}
+
+        # Get values from the loaded answers.conf / passed CLI params
+        for k in result.keys():
+            result[k] = self.config.get(k)
+
+        if bool([a for a in result.values() if a is None]):
+            logger.debug("No required params have been passed")
+            return False
+
+        return True
+
+    def _from_required_params(self):
+        '''
+        Create a default configuration from passed environment parameters. Only PROVIDER_API_KEY
+        is *required*. Token may be blank.
+        '''
+
+        # Initialize the values
+        result = {PROVIDER_API_KEY: self.provider_api,
+                  PROVIDER_AUTH_KEY: self.provider_auth,
+                  PROVIDER_TLS_VERIFY_KEY: self.provider_tls_verify,
+                  PROVIDER_CA_KEY: self.provider_ca}
+
+        # Get values from the loaded answers.conf / passed CLI params
+        for k in result.keys():
+            result[k] = self.config.get(k)
+
+        # Check that the required parameter is passed. If not, error out.
+        for k in [PROVIDER_API_KEY]:
+            if result[k] is None:
+                msg = "Kubernetes API access: You need to set %s in %s or pass it as a CLI param" % (k, ANSWERS_FILE)
```

Comment:
very long line.. would be nice if we go back through at some point and try to keep things under 80 chars